src/trossen_rslidar_main.cpp

```diff
@@ -32,86 +32,22 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #include <signal.h>
 
-#include <manager/node_manager.hpp>
 #include <rclcpp/rclcpp.hpp>
 #include <rs_driver/macro/version.hpp>
-
-using NodeManager = robosense::lidar::NodeManager;
-
-std::mutex g_mtx;
-std::condition_variable g_cv;
-
-static void sigHandler(int sig)
-{
-  RS_MSG << "RoboSense-LiDAR-Driver is stopping....." << RS_REND;
-  g_cv.notify_all();
-}
-
-void show_usage(std::string name)
-{
-  std::cerr << "Usage: " << name << " <option(s)> SOURCES"
-            << "Options:\n"
-            << "\t--config-path\t\tPass the config file absolute path\n"
-            << std::endl;
-}
-
-std::string arg_parser(int argc, char * argv[])
-{
-  std::string path = "";
-  if (argc > 1) {
-    for (int i = 1; i < argc; i++) {
-      std::string arg = argv[i];
-      if (arg == "--config_filepath") {
-        path = argv[i+1];
-        return path;
-      }
-      i++;
-    }
-  } else {
-    std::cerr << "No input given. Resorting to default parameters\n";
-    show_usage(argv[0]);
-  }
-  return path;
-}
+#include <trossen_rslidar_node.hpp>
 
 int main(int argc, char ** argv)
 {
-  signal(SIGINT, sigHandler);  //< bind ctrl+c signal with the sigHandler function
-
-  RS_TITLE << "********************************************************" << RS_REND;
-  RS_TITLE << "**********    RSLidar_SDK Version: v" << RSLIDAR_VERSION_MAJOR
-           << "." << RSLIDAR_VERSION_MINOR
-           << "." << RSLIDAR_VERSION_PATCH << "     **********" << RS_REND;
-  RS_TITLE << "********************************************************" << RS_REND;
-
   rclcpp::init(argc, argv);
+  auto options = rclcpp::NodeOptions();
+  auto node = std::make_shared<robosense::lidar::PointCloudLFNode>(options);
 
-  // Get the YAML Config file path
-  std::string config_path;
-  config_path = arg_parser(argc, argv);
-  if (config_path == "") {
-    config_path = static_cast<std::string>(PROJECT_PATH);
-    config_path += "/config/config.yaml";
-  }
-
-  // Read the YAML file
-  YAML::Node config;
-  try {
-    config = YAML::LoadFile(config_path);
-  } catch (...) {
-    RS_ERROR << "The format of config file " << config_path
-             << " is wrong. Please check (e.g. indentation)." << RS_REND;
-    return -1;
-  }
-
-  std::shared_ptr<NodeManager> demo_ptr = std::make_shared<NodeManager>();
-  demo_ptr->init(config);
-  demo_ptr->start();
-
+  node->configure();
+  node->activate();
```

Comment:
We're creating a lifecycle manager for this node so we don't need to configure and activate it.

Reply:
Left that in so that if we ever load it using the ros2 run command, it can load this executable.

Reply:
We may want to run this as a Node instead of a ComposableNode but still want to be able to manage it using a lifecycle manager. I believe that keeping this will prevent that.